### PR TITLE
Adding initial verify and merge templates

### PR DESCRIPTION
**Pull request by tykeal**
Add the initial verify and merge templates for java projects

Signed-off-by: Andrew Grimberg <agrimberg@linuxfoundation.org>

### CODE REVIEW

**Comment by tykeal** [comment]
@jpwhitemn and @trcox Here's my initial foray into the verify and merge templates for Java projects.

File: jjb/edgex-templates-java.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    scm:
+      - lf-infra-github-scm:
+          url: '{git-clone-url}{github-org}/{project}'
+          refspec: ''
+          branch: '{branch}'
+          submodule-recursive: '{submodule-recursive}'
+          choosing-strategy: default
+          jenkins-ssh-credential: '{jenkins-ssh-credential}'
+
```
Comment (JPWKU) [inline jjb/edgex-templates-java.yaml:74]: No triggers for merge job?

Comment (tykeal) [reply jjb/edgex-templates-java.yaml:74]: Hmm... good catch. I just pushed a new version of it. I'm rather disliking the lost change history here :-/

**Comment by JPWKU** [comment]
LGTM so far but wanna get @zxiiro to look at it too.

File: jjb/edgex-templates-java.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    #
+    # Required Variables:
+    #    branch:   git branch (eg. master)
+
+    <<: *job_boiler_plate
+    # yamllint disable-line rule:key-duplicates
+    <<: *merge_boiler_plate
+
+    # Defaults
+    mvn-goals: 'clean install deploy'
```
Comment (zxiiro) [inline jjb/edgex-templates-java.yaml]: clean deploy

"install" is part of the deploy goal so it's redundant.

File: jjb/edgex-templates-java.yaml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+          maven-version: '{mvn-version}'
+          goals: |
+              {mvn-goals}
+              -Dmerge
+              -Dstream={stream}
+              {edgex-infra-mvn-opts}
+          java-opts:
+            - '{mvn-opts}'
+          properties:
+            - deployAtEnd=true
```
Comment (zxiiro) [inline jjb/edgex-templates-java.yaml:150]: Keep in mind that if any poms in hte project sets `<extensions>true</extensions>` this breaks.

**Review by zxiiro — CHANGES_REQUESTED** [review]
Mainly about `clean deploy`. Fix that and the rest looks good.

File: jjb/edgex-templates-java.yaml (hunk 2 above)
Comment (tykeal) [reply jjb/edgex-templates-java.yaml]: Fixed

File: jjb/edgex-templates-java.yaml (hunk 3 above)
Comment (tykeal) [reply jjb/edgex-templates-java.yaml:150]: None of the projects do currently

**Review by zxiiro — APPROVED** [review]

**Pull request merged**